homeassistant/components/harmony/remote.py

```diff
@@ -81,49 +74,73 @@
             None,
         )
 
-        port = DEFAULT_PORT
-        delay_secs = DEFAULT_DELAY_SECS
-        if override is not None:
-            activity = override.get(ATTR_ACTIVITY)
-            delay_secs = override.get(ATTR_DELAY_SECS)
-            port = override.get(CONF_PORT, DEFAULT_PORT)
+        if matching_configuration_entry is None:
+            # This name is not configured in their yaml
+            return
 
-        host = (discovery_info.get(CONF_NAME), discovery_info.get(CONF_HOST), port)
+        matching_name = discovery_info.get(CONF_NAME)
 
-        # Ignore hub name when checking if this hub is known - ip and port only
-        if host[1:] in ((h.host, h.port) for h in DEVICES):
-            _LOGGER.debug("Discovered host already known: %s", host)
+        # Ignore hub name when checking if this hub is known - ip only
+        if matching_name in (harmony.host for harmony in DEVICES):
+            _LOGGER.debug("Discovered host already known: %s", matching_name)
             return
+
+        import_config = {
+            ATTR_DELAY_SECS: matching_configuration_entry.get(
+                ATTR_DELAY_SECS, DEFAULT_DELAY_SECS
+            ),
+            ATTR_ACTIVITY: matching_configuration_entry.get(ATTR_ACTIVITY),
+            CONF_NAME: matching_name,
+            CONF_HOST: discovery_info.get(CONF_HOST),
+        }
+        _LOGGER.info(
+            "The hub with name '%s' has been found to have address '%s'.",
+            import_config[CONF_NAME],
+            import_config[CONF_HOST],
+        )
+        # The name matches one of the names that was provided in the yaml
+        # We fall though and will proceed to import the config entry
     elif CONF_HOST in config:
-        host = (config.get(CONF_NAME), config.get(CONF_HOST), config.get(CONF_PORT))
-        activity = config.get(ATTR_ACTIVITY)
-        delay_secs = config.get(ATTR_DELAY_SECS)
+        # Here we are loading the yaml config and CONF_HOST is defined
+        import_config = config
     else:
+        # Here we are loading the yaml config and CONF_HOST is NOT defined
+        #
+        # Cache the device config entry so discovery can add it to the list
+        # of entries without CONF_HOST it will auto import as soon as this function
+        # is called with discovery_info that contains a matching CONF_NAME
+        _LOGGER.info(
+            "The hub with name '%s' will be imported upon discovery.",
+            config[CONF_NAME],
+        )
         hass.data[CONF_DEVICE_CACHE].append(config)
         return
 
-    name, address, port = host
-    _LOGGER.info(
-        "Loading Harmony Platform: %s at %s:%s, startup activity: %s",
-        name,
-        address,
-        port,
-        activity,
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
```

Comment:
You should have an import step called here, not init

Reply:
Sorry, I'm not seeing what change to make here.  I'm calling this with `{"source": SOURCE_IMPORT}`

Reply:
Sorry, this is correct